src/assisted_test_infra/download_logs/download_logs.py

```diff
@@ -212,6 +216,76 @@ def download_logs(
 
 
+@JunitTestCase()
+def download_logs_kube_api(
+    api_client: ApiClient,
+    cluster_name: str,
+    namespace: str,
+    dest: str,
+    must_gather: bool,
+):
+
+    cluster_deployment = ClusterDeployment(
+        kube_api_client=api_client,
+        name=cluster_name,
+        namespace=namespace,
+    )
+
+    agent_cluster_install = AgentClusterInstall(
+        kube_api_client=api_client,
+        name=cluster_deployment.get()["spec"]["clusterInstallRef"]["name"],
+        namespace=namespace,
+    )
+
+    output_folder = os.path.join(dest, f"{cluster_name}")
+    recreate_folder(output_folder)
+
+    try:
+        with SuppressAndLog(requests.exceptions.RequestException, ConnectionError):
+            collect_debug_info_from_cluster(cluster_deployment, agent_cluster_install, output_folder)
+
+        if must_gather:
+            recreate_folder(os.path.join(output_folder, "must-gather"))
+            with SuppressAndLog(Exception):
+                # in case of hypershift
+                if namespace.startswith("clusters"):
+                    get_hypershift_logs(api_client, cluster_name, output_folder, namespace)
+                else:
+                    kubeconfig_path = os.path.join(output_folder, "kubeconfig")
+                    agent_cluster_install.download_kubeconfig(kubeconfig_path=kubeconfig_path)
+                    log.info("Agent cluster install spec %s", agent_cluster_install.get()["spec"])
+                    config_etc_hosts(
+                        cluster_name,
+                        cluster_deployment.get()["spec"]["baseDomain"],
+                        agent_cluster_install.get()["spec"]["apiVIP"],
+                    )
+                    download_must_gather(kubeconfig_path, os.path.join(output_folder, "must-gather"))
+
+    finally:
+        run_command(f"chmod -R ugo+rx '{output_folder}'")
+
+
+def get_hypershift_logs(api_client, cluster_name, output_folder, namespace):
+    hypershift = HyperShift(name=cluster_name)
+    kubeconfig_path = hypershift.download_kubeconfig(kube_api_client=api_client)
+    with SuppressAndLog(Exception):
+        log.info("Dumping hypershift files")
+        hypershift.dump(os.path.join(output_folder, "dump"))
+    with SuppressAndLog(Exception):
+        output_folder = os.path.join(output_folder, "must-gather")
+        recreate_folder(output_folder)
+        command = (
+            f"oc --kubeconfig {kubeconfig_path} get pods --all-namespaces  "
+            f"-o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name   --no-headers "
+            f"--no-headers | xargs -n 2 | cut -d' ' -f1,2 "
+            f"| xargs -n 2 bash -c 'echo getting logs of $1 in $0; "
+            f"oc --kubeconfig {kubeconfig_path} logs $1 -n $0  "
+            f"--all-containers > {output_folder}/logs_$1_hypershift.log'"
+        )
+        log.info("Getting all cluster logs with %s", command)
+        utils.run_command_with_output(command)
```

Comment:
Why do we need this part?
Also, perhaps this should get placed under cluster specific folder ({output_folder}/hypershift/{namespace}/$1.log'?

Reply:
this parts downloads all the logs from hypershift hosted cluster as must gather is not working. If we don't need it , will be great to  remove